its/src/test/java/com/sonar/it/scanner/msbuild/ScannerMSBuildTest.java

```diff
@@ -816,6 +816,61 @@ public void testIgnoreIssuesDoesNotRemoveSourceGenerator() throws IOException {
     assertThat(filter(issues, ROSLYN_RULES_PREFIX)).isEmpty();
   }
 
+  @Test
+  public void whenEachProjectIsOnDifferentDrives_AnalysisFails() throws IOException {
+    Path projectDir = TestUtils.projectDir(temp, "TwoDrivesTwoProjects");
+
+    int setupStatus = CommandExecutor.create().execute(
+      Command.create("setup.bat")
+        .addArguments(projectDir.resolve( "DriveZ").toAbsolutePath().toString())
+        .setDirectory(projectDir.toFile()),
+      60 * 1000);
+    assertThat(setupStatus).isZero();
+
+    BuildResult buildResult = runAnalysisWithoutProjectBasedDir(projectDir);
+
+    assertThat(buildResult.isSuccess()).isFalse();
+    assertThat(buildResult.getLogs()).contains("Generation of the sonar-properties file failed. Unable to complete the analysis.");
+
+    int cleanupStatus = CommandExecutor.create().execute(Command.create("cleanup.bat").setDirectory(projectDir.toFile()), 60 * 1000);
+    assertThat(cleanupStatus).isZero();
+  }
+
+  @Test
+  public void whenMinorityOfProjectsIsOnDifferentDrives_AnalysisSucceeds() throws IOException {
+    Path projectDir = TestUtils.projectDir(temp, "TwoDrivesThreeProjects");
+    int setupStatus = CommandExecutor.create().execute(
+      Command.create("setup.bat")
+        .addArguments(projectDir.resolve( "DriveY").toAbsolutePath().toString())
+        .setDirectory(projectDir.toFile()),
+      60 * 1000);
+    assertThat(setupStatus).isZero();
+
+    BuildResult buildResult = runAnalysisWithoutProjectBasedDir(projectDir);
+
+    assertThat(buildResult.isSuccess()).isTrue();
+    assertThat(buildResult.getLogs()).contains("Using longest common projects path as a base directory: '" + projectDir);
+    assertThat(buildResult.getLogs()).contains("WARNING: Directory 'Y:\\' is not located under the base directory '" + projectDir + "' and will not be analyzed.");
+    assertThat(buildResult.getLogs()).contains("WARNING: File 'Y:\\Program.cs' is not located under the base directory '" + projectDir + "' and will not be analyzed.");
+    assertThat(buildResult.getLogs()).contains("File was referenced by the following projects: 'Y:\\DriveY.csproj'.");
+
+    int cleanupStatus = CommandExecutor.create().execute(Command.create("cleanup.bat").setDirectory(projectDir.toFile()), 60 * 1000);
+    assertThat(cleanupStatus).isZero();
+  }
+
+  @Test
+  public void testAzureFunctions_AnalysisSuceeds_WithWrongBaseDirectory() throws IOException {
```

Comment:
To align with the previous test name for consistency
```suggestion
  public void testAzureFunctions_WithWrongBaseDirectory_AnalysisSuceeds() throws IOException {
```